### ISSUE TITLE
Armory 0.6: cannot start Krom on macOS

### ISSUE THREAD

**Issue by robw001** [armory3d/armory#1190]
SDK version: both bundled in 0.6 distribution and latest from Git
Blender version: 2.8
OS: macOS 10.13.6

Starting Krom from within blender 2.8 exports/compiles the project but does not start Krom. The compile run has the following at the end of the output:

> Finished in 25.14314603805542
> deviceBufferSize = 12288
>
> mSampleRate = 48000
>
> mFormatFlags = 00000009
> 
> mBytesPerPacket = 24
> 
> mFramesPerPacket = 1
> 
> mChannelsPerFrame = 6
> 
> mBytesPerFrame = 24
> 
> mBitsPerChannel = 32

After this, things just seem to disappear into /dev/null…

Building a project and then starting Krom from a terminal window works. (see discussion [here](http://forums.armory3d.org/t/issue-with-armory-0-6-and-custom-nodes/2729/11))

**Comment by Bugsbane** [comment]
This might be a macOS specific issue, as both 0.6 standard download and current git ( a6c954cda7ca429e3887069100e11ba5441b88c9 ) start Krom fine for me on Kubuntu 64 bit 18.10.

**Comment by BlackGoku36** [comment]
@luboslenco just found out that Macos doesn't crash because of bug, it crash because of cache, clicking `Clean` few time and then running again solve the problem

**Comment by sakthig** [comment]
@BlackGoku36 . Still crashes for me even if i clean the proj multiple times. But if i publish and run the executable that crashes some times but works fine most of the times.
[game.blend.zip](https://github.com/armory3d/armory/files/3194097/game.blend.zip)

**Comment by lewislepton** [comment]
I too am getting this crash. even from a very simple stater example that blender generates.
tried shooting any errors, but terminal has none. only the build log, which is fine, much like kha. it just opens & then closes up krom right away.

I'm on current public version of macOS Mojave 14.4 using the 0.6 armory3D

**Comment by masondesu** [comment]
This may be a red herring since multiple issues were closed and redirected to this issue, but I've been seeing similar problems to what's described here and in other issues where Krom would quit unexpectedly a few seconds after pressing Play with the Krom runtime in Armory 0.6 and on the current armsdk master (https://github.com/armory3d/armsdk/commit/6b2905366ae31e67cacef25cd7e1716769f571fb) on macOS Mojave 10.14.4. 

The build / publish flow worked fine and the generated Krom executable ran fine, and the browser runtime always worked. I dug in and found that the `--sound` flag that's added at the play command at https://github.com/armory3d/armory/blob/master/blender/arm/make.py#L523 causes the Krom to quit unexpectedly. Removing this line makes the Play button behave exactly as expected, both for the Playground demo and fresh blend files made on my machine. My examples aren't attempting to play sound effects, I'm assuming that removing this flag does indeed disable sound.

I'm happy to open a pull request, but I'm not clear what the fix is. I saw that this was recently touched here on Krom https://github.com/Kode/Krom/pull/123 by @luboslenco, so perhaps they have some insight into the fix. Thanks so much!
